### PR TITLE
Run3-gex138B Replace access by label with access by token in CalibCalorimetry/EcalCorrelatedNoiseAnalysisModules

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Replace access by label with access by token in CalibCalorimetry/EcalCorrelatedNoiseAnalysisModules

#### PR validation:

Use the runTheMatrix test workflows

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: CalibCalorimetry/EcalCorrelatedNoiseAnalysisModules/src/EcnaAnalyzer.cc [hunk 1]
```diff
@@ -1409,20 +1383,18 @@ void EcnaAnalyzer::analyze(const edm::Event &iEvent, const edm::EventSetup &iSet
...
-      //<< " i_current_ev_time = " << i_current_ev_time  << std::endl
-      //<< " p_current_ev_time = " << p_current_ev_time  << std::endl
-      // std::cout 	    << " Last analyzed event date  = " << astime <<
-      // std::endl;
-      std::cout << " Number of selected events = " << fNbOfSelectedEvents << std::endl;
-      std::cout << std::endl
-                << fNbOfTreatedStexs << " " << fStexName << "'s with " << fReqNbOfEvts << " events analyzed."
-                << std::endl
-                << "---------------------------------------------------------" << std::endl;
+      // edm::LogVerbatim("ecnaAnal") << " t_current_ev_time = " << t_current_ev_time  <<
```
Comment (ChrisMisan) [inline CalibCalorimetry/EcalCorrelatedNoiseAnalysisModules/src/EcnaAnalyzer.cc:1389]: Is this code needed?

**Review by ChrisMisan — COMMENTED** [review]
@bsunanda Additional comments:

- floating namespaces, please consider using namespace prefix or put the namespace in the top section of the file to avoid confusion
https://github.com/cms-sw/cmssw/blob/d0785b767d61438ed2a6796ab10df40101d0dd07/CalibCalorimetry/EcalCorrelatedNoiseAnalysisModules/src/EcnaAnalyzer.cc#L47

- this should be a switch or it can be collapsed to a loop
https://github.com/cms-sw/cmssw/blob/d0785b767d61438ed2a6796ab10df40101d0dd07/CalibCalorimetry/EcalCorrelatedNoiseAnalysisModules/src/EcnaAnalyzer.cc#L1648

File: CalibCalorimetry/EcalCorrelatedNoiseAnalysisModules/src/EcnaAnalyzer.cc [hunk 2]
```diff
@@ -53,29 +53,25 @@ EcnaAnalyzer::EcnaAnalyzer(const edm::ParameterSet &pSet)
...
 
-  std::cout << "*EcnaAnalyzer-constructor> Check path for resultsq Root files." << std::endl;
+  edm::LogVerbatim("ecnaAnal") << "*EcnaAnalyzer-constructor> Check path for resultsq Root files.";
 
   if (myPathEB->GetPathForResultsRootFiles() == kFALSE) {
-    std::cout << "*EcnaAnalyzer-constructor> *** ERROR *** Path for result "
-                 "files not found."
-              << std::endl;
+    edm::LogVerbatim("ecnaAnal") << "*EcnaAnalyzer-constructor> *** ERROR *** Path for result files not found.";
     kill(getpid(), SIGUSR2);
```
[the remark this answers is not in the document]
Comment (tvami) [reply CalibCalorimetry/EcalCorrelatedNoiseAnalysisModules/src/EcnaAnalyzer.cc:60]: Yeah, that `kill(getpid(), SIGUSR2);` is something weird, which shouldnt really happen.

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: CalibCalorimetry/EcalCorrelatedNoiseAnalysisModules/src/EcnaAnalyzer.cc [hunk 3]
```diff
@@ -38,44 +39,37 @@
...
+    edm::LogVerbatim("ecnaAnal") << "*EcnaAnalyzer-constructor> Path for result files found = "
+                                 << myPathEB->ResultsRootFilePath();
   }
 
   if (myPathEE->GetPathForResultsRootFiles() == kFALSE) {
-    std::cout << "*EcnaAnalyzer-constructor> *** ERROR *** Path for result "
-                 "files not found."
-              << std::endl;
+    edm::LogVerbatim("ecnaAnal") << "*EcnaAnalyzer-constructor> *** ERROR *** Path for result files not found.";
     kill(getpid(), SIGUSR2);
```
Comment (ChrisMisan) [inline CalibCalorimetry/EcalCorrelatedNoiseAnalysisModules/src/EcnaAnalyzer.cc:66]: this should be addressed as well

File: CalibCalorimetry/EcalCorrelatedNoiseAnalysisModules/src/EcnaAnalyzer.cc [hunk 4]
```diff
@@ -1501,114 +1444,75 @@ Bool_t EcnaAnalyzer::AnalysisOutcome(const TString &s_opt) {
...
-                << fLastReqEvent << std::endl
-                << "                                     Current event number  "
-                   "      = "
-                << fCurrentEventNumber << std::endl;
+      edm::LogVerbatim("ecnaAnal")
+          << "\n**************************** ANALYSIS REPORT > OK **************************************"
+          << "\n*EcnaAnalyzer::AnalysisOutcome(...)> The maximum requested number of events and the maximum"
+          << "\n                                     number of treated " << fStexName << "'s have been reached."
+          << "\n                                     Analysis successfully ended from EcnaAnalyzer "
+          << "\n                                     by SIGNAL: kill(getpid(),SIGUSR2)."
```
Comment (ChrisMisan) [inline CalibCalorimetry/EcalCorrelatedNoiseAnalysisModules/src/EcnaAnalyzer.cc:1452]: this should be changed

File: CalibCalorimetry/EcalCorrelatedNoiseAnalysisModules/src/EcnaAnalyzer.cc [hunk 5]
```diff
@@ -1501,114 +1444,75 @@ Bool_t EcnaAnalyzer::AnalysisOutcome(const TString &s_opt) {
...
-                << fLastReqEvent << std::endl
-                << "                                     Current event number  "
-                   "      = "
-                << fCurrentEventNumber << std::endl;
+      edm::LogVerbatim("ecnaAnal") << "\n**************************** ANALYSIS REPORT >>> *** "
+                                      "WARNING *** WARNING *** WARNING ***"
+                                   << "\n*EcnaAnalyzer::AnalysisOutcome(...)> Last event reached "
+                                      "before completion of analysis."
+                                   << "\n                                     Analysis ended from EcnaAnalyzer "
+                                   << "\n                                     by SIGNAL: kill(getpid(),SIGUSR2)."
```
Comment (ChrisMisan) [inline CalibCalorimetry/EcalCorrelatedNoiseAnalysisModules/src/EcnaAnalyzer.cc:1474]: this should be changed

File: CalibCalorimetry/EcalCorrelatedNoiseAnalysisModules/src/EcnaAnalyzer.cc [hunk 6]
```diff
@@ -1501,114 +1444,75 @@ Bool_t EcnaAnalyzer::AnalysisOutcome(const TString &s_opt) {
...
 
-      std::cout << "***********************************************************"
-                   "*****************************"
-                << std::endl
-                << std::endl;
+      edm::LogVerbatim("ecnaAnal")
+          << "****************************************************************************************" << std::endl;
 
       result = kTRUE;
       kill(getpid(), SIGUSR2);
```
Comment (ChrisMisan) [inline CalibCalorimetry/EcalCorrelatedNoiseAnalysisModules/src/EcnaAnalyzer.cc:1489]: this should be addressed as well

File: CalibCalorimetry/EcalCorrelatedNoiseAnalysisModules/src/EcnaAnalyzer.cc [hunk 7]
```diff
@@ -1501,114 +1444,75 @@ Bool_t EcnaAnalyzer::AnalysisOutcome(const TString &s_opt) {
...
-      std::cout << "***********************************************************"
-                   "*****************************"
-                << std::endl
-                << std::endl;
+      edm::LogVerbatim("ecnaAnal")
+          << "\n**************************** ANALYSIS REPORT >>> **** ERROR **** ERROR **** ERROR ******"
+          << "\n*EcnaAnalyzer::AnalysisOutcome(...)> First event number = " << fFirstReqEvent
+          << ". Should be strictly potitive."
+          << "\n                             Analysis ended from EcnaAnalyzer "
+          << "\n                             by SIGNAL: kill(getpid(),SIGUSR2).";
```
Comment (ChrisMisan) [inline CalibCalorimetry/EcalCorrelatedNoiseAnalysisModules/src/EcnaAnalyzer.cc:1498]: this should be changed

File: CalibCalorimetry/EcalCorrelatedNoiseAnalysisModules/src/EcnaAnalyzer.cc [hunk 8]
```diff
@@ -1501,114 +1444,75 @@ Bool_t EcnaAnalyzer::AnalysisOutcome(const TString &s_opt) {
...
+
+      edm::LogVerbatim("ecnaAnal")
+          << "****************************************************************************************" << std::endl;
 
       result = kTRUE;
       kill(getpid(), SIGUSR2);
```
Comment (ChrisMisan) [inline CalibCalorimetry/EcalCorrelatedNoiseAnalysisModules/src/EcnaAnalyzer.cc:1504]: this should be addressed as well

File: CalibCalorimetry/EcalCorrelatedNoiseAnalysisModules/src/EcnaAnalyzer.cc [hunk 9]
```diff
@@ -1501,114 +1444,75 @@ Bool_t EcnaAnalyzer::AnalysisOutcome(const TString &s_opt) {
...
+          << "\n*EcnaAnalyzer::analyze(...)> Requested number of events = " << fReqNbOfEvts << "."
+          << "\n                             Too large compared to the event range: " << fFirstReqEvent << " - "
+          << fLastReqEvent << "\n                             Analysis ended from EcnaAnalyzer "
+          << "\n                             by SIGNAL: kill(getpid(),SIGUSR2).";
+
+      edm::LogVerbatim("ecnaAnal")
+          << "****************************************************************************************" << std::endl;
 
       result = kTRUE;
       kill(getpid(), SIGUSR2);
```
Comment (ChrisMisan) [inline CalibCalorimetry/EcalCorrelatedNoiseAnalysisModules/src/EcnaAnalyzer.cc:1518]: this should be addressed as well

File: CalibCalorimetry/EcalCorrelatedNoiseAnalysisModules/src/EcnaAnalyzer.cc [hunk 10]
```diff
@@ -1501,114 +1444,75 @@ Bool_t EcnaAnalyzer::AnalysisOutcome(const TString &s_opt) {
...
 
-      std::cout << "***********************************************************"
-                   "*****************************"
-                << std::endl
-                << std::endl;
+      edm::LogVerbatim("ecnaAnal")
+          << "****************************************************************************************\n";
 
       result = kTRUE;
       kill(getpid(), SIGUSR2);
```
Comment (ChrisMisan) [inline CalibCalorimetry/EcalCorrelatedNoiseAnalysisModules/src/EcnaAnalyzer.cc:1464]: this should be addressed as well

File: CalibCalorimetry/EcalCorrelatedNoiseAnalysisModules/src/EcnaAnalyzer.cc [hunk 11]
```diff
@@ -1501,114 +1444,75 @@ Bool_t EcnaAnalyzer::AnalysisOutcome(const TString &s_opt) {
...
-      std::cout << "***********************************************************"
-                   "*****************************"
-                << std::endl
-                << std::endl;
+      edm::LogVerbatim("ecnaAnal")
+          << "\n**************************** ANALYSIS REPORT >>> **** ERROR **** ERROR **** ERROR ******"
+          << "\n*EcnaAnalyzer::analyze(...)> Requested number of events = " << fReqNbOfEvts << "."
+          << "\n                             Too large compared to the event range: " << fFirstReqEvent << " - "
+          << fLastReqEvent << "\n                             Analysis ended from EcnaAnalyzer "
+          << "\n                             by SIGNAL: kill(getpid(),SIGUSR2).";
```
Comment (ChrisMisan) [inline CalibCalorimetry/EcalCorrelatedNoiseAnalysisModules/src/EcnaAnalyzer.cc:1512]: this should be changed

**Review by ChrisMisan — COMMENTED** [review]
@bsunanda memory management issue and kill issue were not resolved for the whole file

**Comment by ChrisMisan** [comment]
hold

- holding until memory leak issues are fixed

**Comment by ChrisMisan** [comment]
@bsunanda kind reminder

**Comment by bsunanda** [comment]
@chrismisan Please indicate what else to be done to get this accepted.

**Comment by ChrisMisan** [comment]
> @ChrisMisan Please indicate what else to be done to get this accepted.

@bsunanda I already did: https://github.com/cms-sw/cmssw/pull/39207#pullrequestreview-1087670237

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by ChrisMisan** [comment]
please abort

- memory deallocation issues were not resolved

**Comment by ChrisMisan** [comment]
@bsunanda Only one part of the requested changes were implemented. There are many instances were memory is not properly deallocated in this file. As a rule of thumb every new-ed memory should be deleted.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by ChrisMisan** [comment]
@bsunanda in the future it'd be much better to have smart pointers in this file, instead of manual memory management.

**Comment by ChrisMisan** [comment]
+alca

- differences in L1T and MTD are not related to this PR
- technical PR

**Comment by bsunanda** [comment]
@ChrisMisan Please uphold this. Let the modernizining the code be done in a later PR

**Comment by ChrisMisan** [comment]
unhold

**Comment by perrotta** [comment]
+1

**Pull request merged**